src/utils.hpp

```diff
@@ -0,0 +1,34 @@
+#if 0
+#pragma once
+#include <cstddef>
+#include <string.h>
+
+namespace utils {
+
+	bool safe_strn_copy(char *dest, size_t destSize, const char *src) {
+		//Code from: https://stackoverflow.com/a/63331927
+		// Optional pointer test
+		if (dest == NULL) return true;
```

Comment:
Don't use NULL, use nullptr instead. nullptr can't be implicitly converted to an integral type.

Reply:
Code from stack overflow, I'll remove it all, yup, good refresher of nullptr btw